{
  "cells": [
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "view-in-github",
        "colab_type": "text"
      },
      "source": [
        "<a href=\"https://colab.research.google.com/github/martings/Topicos_Bioinfo/blob/main/Notebook_0_FastQ_Descarga_EDA_Trimming_y_multiQC.ipynb\" target=\"_parent\"><img src=\"https://colab.research.google.com/assets/colab-badge.svg\" alt=\"Open In Colab\"/></a>"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "8CNMeMDT_MdS"
      },
      "source": [
        "El control de calidad (QC) de los datos de secuenciación es un paso crucial en los análisis bioinformáticos. Los archivos FastQ, que contienen las secuencias de nucleótidos y sus respectivas calidades, deben ser evaluados para asegurar que los datos son de alta calidad antes de proceder con análisis posteriores. [FastQC](https://www.bioinformatics.babraham.ac.uk/projects/fastqc/) es una herramienta ampliamente utilizada para generar informes detallados sobre la calidad de los datos de secuenciación.\n",
        "\n",
        "[MultiQC](https://multiqc.info/) es una herramienta que agrega múltiples informes de FastQC en un solo reporte comprensible."
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 1,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 1000
        },
        "id": "Ea6arN2vrW0S",
        "outputId": "ab3bde6b-1378-4dc8-eb1c-db088cd2f951"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "\r0% [Working]\r            \rGet:1 https://cloud.r-project.org/bin/linux/ubuntu jammy-cran40/ InRelease [3,626 B]\n",
            "Hit:2 https://developer.download.nvidia.com/compute/cuda/repos/ubuntu2204/x86_64  InRelease\n",
            "Get:3 http://security.ubuntu.com/ubuntu jammy-security InRelease [129 kB]\n",
            "Hit:4 http://archive.ubuntu.com/ubuntu jammy InRelease\n",
            "Get:5 http://archive.ubuntu.com/ubuntu jammy-updates InRelease [128 kB]\n",
            "Hit:6 https://ppa.launchpadcontent.net/c2d4u.team/c2d4u4.0+/ubuntu jammy InRelease\n",
            "Hit:7 https://ppa.launchpadcontent.net/deadsnakes/ppa/ubuntu jammy InRelease\n",
            "Hit:8 https://ppa.launchpadcontent.net/graphics-drivers/ppa/ubuntu jammy InRelease\n",
            "Hit:9 http://archive.ubuntu.com/ubuntu jammy-backports InRelease\n",
            "Hit:10 https://ppa.launchpadcontent.net/ubuntugis/ppa/ubuntu jammy InRelease\n",
            "Get:11 http://archive.ubuntu.com/ubuntu jammy-updates/universe amd64 Packages [1,392 kB]\n",
            "Fetched 1,652 kB in 3s (483 kB/s)\n",
            "Reading package lists... Done\n",
            "Reading package lists... Done\n",
            "Building dependency tree... Done\n",
            "Reading state information... Done\n",
            "The following additional packages will be installed:\n",
            "  blends-common libkdf5-2 libncbi-vdb2 libncbi-wvdb2 med-config menu\n",
            "Suggested packages:\n",
            "  blends-doc menu-l10n gksu | kde-runtime | ktsuss\n",
            "The following NEW packages will be installed:\n",
            "  blends-common libkdf5-2 libncbi-vdb2 libncbi-wvdb2 med-config menu sra-toolkit\n",
            "0 upgraded, 7 newly installed, 0 to remove and 45 not upgraded.\n",
            "Need to get 8,290 kB of archives.\n",
            "After this operation, 23.0 MB of additional disk space will be used.\n",
            "Get:1 http://archive.ubuntu.com/ubuntu jammy/universe amd64 menu amd64 2.1.47ubuntu4 [354 kB]\n",
            "Get:2 http://archive.ubuntu.com/ubuntu jammy/universe amd64 blends-common all 0.7.4ubuntu1 [15.9 kB]\n",
            "Get:3 http://archive.ubuntu.com/ubuntu jammy/universe amd64 libkdf5-2 amd64 2.11.2+dfsg-4build2 [14.7 kB]\n",
            "Get:4 http://archive.ubuntu.com/ubuntu jammy/universe amd64 libncbi-vdb2 amd64 2.11.2+dfsg-4build2 [1,364 kB]\n",
            "Get:5 http://archive.ubuntu.com/ubuntu jammy/universe amd64 libncbi-wvdb2 amd64 2.11.2+dfsg-4build2 [1,252 kB]\n",
            "Get:6 http://archive.ubuntu.com/ubuntu jammy/universe amd64 sra-toolkit amd64 2.11.3+dfsg-1ubuntu1 [5,277 kB]\n",
            "Get:7 http://archive.ubuntu.com/ubuntu jammy/universe amd64 med-config all 3.7.1 [11.6 kB]\n",
            "Fetched 8,290 kB in 2s (4,458 kB/s)\n",
            "Preconfiguring packages ...\n",
            "Selecting previously unselected package menu.\n",
            "(Reading database ... 121925 files and directories currently installed.)\n",
            "Preparing to unpack .../0-menu_2.1.47ubuntu4_amd64.deb ...\n",
            "Unpacking menu (2.1.47ubuntu4) ...\n",
            "Selecting previously unselected package blends-common.\n",
            "Preparing to unpack .../1-blends-common_0.7.4ubuntu1_all.deb ...\n",
            "Unpacking blends-common (0.7.4ubuntu1) ...\n",
            "Selecting previously unselected package libkdf5-2.\n",
            "Preparing to unpack .../2-libkdf5-2_2.11.2+dfsg-4build2_amd64.deb ...\n",
            "Unpacking libkdf5-2 (2.11.2+dfsg-4build2) ...\n",
            "Selecting previously unselected package libncbi-vdb2.\n",
            "Preparing to unpack .../3-libncbi-vdb2_2.11.2+dfsg-4build2_amd64.deb ...\n",
            "Unpacking libncbi-vdb2 (2.11.2+dfsg-4build2) ...\n",
            "Selecting previously unselected package libncbi-wvdb2.\n",
            "Preparing to unpack .../4-libncbi-wvdb2_2.11.2+dfsg-4build2_amd64.deb ...\n",
            "Unpacking libncbi-wvdb2 (2.11.2+dfsg-4build2) ...\n",
            "Selecting previously unselected package sra-toolkit.\n",
            "Preparing to unpack .../5-sra-toolkit_2.11.3+dfsg-1ubuntu1_amd64.deb ...\n",
            "Unpacking sra-toolkit (2.11.3+dfsg-1ubuntu1) ...\n",
            "Selecting previously unselected package med-config.\n",
            "Preparing to unpack .../6-med-config_3.7.1_all.deb ...\n",
            "Unpacking med-config (3.7.1) ...\n",
            "Setting up libncbi-vdb2 (2.11.2+dfsg-4build2) ...\n",
            "Setting up libkdf5-2 (2.11.2+dfsg-4build2) ...\n",
            "Setting up libncbi-wvdb2 (2.11.2+dfsg-4build2) ...\n",
            "Setting up menu (2.1.47ubuntu4) ...\n",
            "Setting up blends-common (0.7.4ubuntu1) ...\n",
            "Setting up sra-toolkit (2.11.3+dfsg-1ubuntu1) ...\n",
            "Setting up med-config (3.7.1) ...\n",
            "Adding group `med' (GID 107) ...\n",
            "Done.\n",
            "Processing triggers for man-db (2.10.2-1) ...\n",
            "Processing triggers for libc-bin (2.35-0ubuntu3.4) ...\n",
            "/sbin/ldconfig.real: /usr/local/lib/libtbbbind_2_0.so.3 is not a symbolic link\n",
            "\n",
            "/sbin/ldconfig.real: /usr/local/lib/libtbbmalloc_proxy.so.2 is not a symbolic link\n",
            "\n",
            "/sbin/ldconfig.real: /usr/local/lib/libtbbmalloc.so.2 is not a symbolic link\n",
            "\n",
            "/sbin/ldconfig.real: /usr/local/lib/libtbbbind.so.3 is not a symbolic link\n",
            "\n",
            "/sbin/ldconfig.real: /usr/local/lib/libtbb.so.12 is not a symbolic link\n",
            "\n",
            "/sbin/ldconfig.real: /usr/local/lib/libtbbbind_2_5.so.3 is not a symbolic link\n",
            "\n",
            "Processing triggers for menu (2.1.47ubuntu4) ...\n",
            "Reading package lists... Done\n",
            "Building dependency tree... Done\n",
            "Reading state information... Done\n",
            "The following additional packages will be installed:\n",
            "  default-jre default-jre-headless fonts-dejavu-core fonts-dejavu-extra libapache-pom-java\n",
            "  libargs4j-java libatk-wrapper-java libatk-wrapper-java-jni libcommons-compress-java\n",
            "  libcommons-io-java libcommons-jexl2-java libcommons-lang3-java libcommons-logging-java\n",
            "  libcommons-math3-java libcommons-parent-java libfindbin-libs-perl libfontenc1 libhtsjdk-java\n",
            "  libjbzip2-java libjson-simple-java libngs-java libngs-sdk-dev libngs-sdk2 libsis-base-java\n",
            "  libsis-base-jni libsis-jhdf5-java libsis-jhdf5-jni libsnappy-java libsnappy-jni libxkbfile1\n",
            "  libxtst6 libxxf86dga1 openjdk-11-jre x11-utils\n",
            "Suggested packages:\n",
            "  libxz-java libcommons-io-java-doc libcommons-lang3-java-doc libavalon-framework-java\n",
            "  libcommons-logging-java-doc libexcalibur-logkit-java liblog4j1.2-java picard-tools\n",
            "  libjson-simple-doc mesa-utils\n",
            "The following NEW packages will be installed:\n",
            "  default-jre default-jre-headless fastqc fonts-dejavu-core fonts-dejavu-extra libapache-pom-java\n",
            "  libargs4j-java libatk-wrapper-java libatk-wrapper-java-jni libcommons-compress-java\n",
            "  libcommons-io-java libcommons-jexl2-java libcommons-lang3-java libcommons-logging-java\n",
            "  libcommons-math3-java libcommons-parent-java libfindbin-libs-perl libfontenc1 libhtsjdk-java\n",
            "  libjbzip2-java libjson-simple-java libngs-java libngs-sdk-dev libngs-sdk2 libsis-base-java\n",
            "  libsis-base-jni libsis-jhdf5-java libsis-jhdf5-jni libsnappy-java libsnappy-jni libxkbfile1\n",
            "  libxtst6 libxxf86dga1 openjdk-11-jre x11-utils\n",
            "0 upgraded, 35 newly installed, 0 to remove and 45 not upgraded.\n",
            "Need to get 11.8 MB of archives.\n",
            "After this operation, 24.8 MB of additional disk space will be used.\n",
            "Get:1 http://archive.ubuntu.com/ubuntu jammy/main amd64 default-jre-headless amd64 2:1.11-72build2 [3,042 B]\n",
            "Get:2 http://archive.ubuntu.com/ubuntu jammy/main amd64 libxtst6 amd64 2:1.2.3-1build4 [13.4 kB]\n",
            "Get:3 http://archive.ubuntu.com/ubuntu jammy-updates/main amd64 openjdk-11-jre amd64 11.0.23+9-1ubuntu1~22.04.1 [214 kB]\n",
            "Get:4 http://archive.ubuntu.com/ubuntu jammy/main amd64 default-jre amd64 2:1.11-72build2 [896 B]\n",
            "Get:5 http://archive.ubuntu.com/ubuntu jammy/universe amd64 libcommons-math3-java all 3.6.1-3 [2,016 kB]\n",
            "Get:6 http://archive.ubuntu.com/ubuntu jammy/universe amd64 libjbzip2-java all 0.9.1-7 [42.8 kB]\n",
            "Get:7 http://archive.ubuntu.com/ubuntu jammy/universe amd64 libapache-pom-java all 18-1 [4,720 B]\n",
            "Get:8 http://archive.ubuntu.com/ubuntu jammy/universe amd64 libcommons-parent-java all 43-1 [10.8 kB]\n",
            "Get:9 http://archive.ubuntu.com/ubuntu jammy/universe amd64 libcommons-compress-java all 1.21-1 [575 kB]\n",
            "Get:10 http://archive.ubuntu.com/ubuntu jammy/universe amd64 libcommons-logging-java all 1.2-2 [60.3 kB]\n",
            "Get:11 http://archive.ubuntu.com/ubuntu jammy/universe amd64 libcommons-jexl2-java all 2.1.1-5 [251 kB]\n",
            "Get:12 http://archive.ubuntu.com/ubuntu jammy/universe amd64 libjson-simple-java all 2.3.0-1 [61.0 kB]\n",
            "Get:13 http://archive.ubuntu.com/ubuntu jammy/universe amd64 libngs-sdk2 amd64 2.11.2-3 [65.6 kB]\n",
            "Get:14 http://archive.ubuntu.com/ubuntu jammy/universe amd64 libngs-sdk-dev amd64 2.11.2-3 [111 kB]\n",
            "Get:15 http://archive.ubuntu.com/ubuntu jammy/universe amd64 libngs-java amd64 2.11.2-3 [75.8 kB]\n",
            "Get:16 http://archive.ubuntu.com/ubuntu jammy/universe amd64 libsnappy-jni amd64 1.1.8.3-1 [6,756 B]\n",
            "Get:17 http://archive.ubuntu.com/ubuntu jammy/universe amd64 libsnappy-java all 1.1.8.3-1 [93.9 kB]\n",
            "Get:18 http://archive.ubuntu.com/ubuntu jammy/universe amd64 libhtsjdk-java all 2.24.1+dfsg-1 [1,644 kB]\n",
            "Get:19 http://archive.ubuntu.com/ubuntu jammy/universe amd64 libfindbin-libs-perl all 2.190.02-1 [23.2 kB]\n",
            "Get:20 http://archive.ubuntu.com/ubuntu jammy/universe amd64 libargs4j-java all 2.33-1.1 [137 kB]\n",
            "Get:21 http://archive.ubuntu.com/ubuntu jammy/universe amd64 libcommons-io-java all 2.11.0-2 [297 kB]\n",
            "Get:22 http://archive.ubuntu.com/ubuntu jammy/universe amd64 libcommons-lang3-java all 3.11-1 [526 kB]\n",
            "Get:23 http://archive.ubuntu.com/ubuntu jammy/universe amd64 libsis-base-jni amd64 18.09~pre1+git20180928.45fbd31+dfsg-3 [12.7 kB]\n",
            "Get:24 http://archive.ubuntu.com/ubuntu jammy/universe amd64 libsis-base-java all 18.09~pre1+git20180928.45fbd31+dfsg-3 [96.4 kB]\n",
            "Get:25 http://archive.ubuntu.com/ubuntu jammy/universe amd64 libsis-jhdf5-jni amd64 19.04.0+dfsg-4 [96.6 kB]\n",
            "Get:26 http://archive.ubuntu.com/ubuntu jammy/universe amd64 libsis-jhdf5-java all 19.04.0+dfsg-4 [1,490 kB]\n",
            "Get:27 http://archive.ubuntu.com/ubuntu jammy/universe amd64 fastqc all 0.11.9+dfsg-5 [370 kB]\n",
            "Get:28 http://archive.ubuntu.com/ubuntu jammy/main amd64 fonts-dejavu-core all 2.37-2build1 [1,041 kB]\n",
            "Get:29 http://archive.ubuntu.com/ubuntu jammy/main amd64 fonts-dejavu-extra all 2.37-2build1 [2,041 kB]\n",
            "Get:30 http://archive.ubuntu.com/ubuntu jammy/main amd64 libfontenc1 amd64 1:1.1.4-1build3 [14.7 kB]\n",
            "Get:31 http://archive.ubuntu.com/ubuntu jammy/main amd64 libxkbfile1 amd64 1:1.1.0-1build3 [71.8 kB]\n",
            "Get:32 http://archive.ubuntu.com/ubuntu jammy/main amd64 libxxf86dga1 amd64 2:1.1.5-0ubuntu3 [12.6 kB]\n",
            "Get:33 http://archive.ubuntu.com/ubuntu jammy/main amd64 x11-utils amd64 7.7+5build2 [206 kB]\n",
            "Get:34 http://archive.ubuntu.com/ubuntu jammy/main amd64 libatk-wrapper-java all 0.38.0-5build1 [53.1 kB]\n",
            "Get:35 http://archive.ubuntu.com/ubuntu jammy/main amd64 libatk-wrapper-java-jni amd64 0.38.0-5build1 [49.0 kB]\n",
            "Fetched 11.8 MB in 2s (5,300 kB/s)\n",
            "Extracting templates from packages: 100%\n",
            "Selecting previously unselected package default-jre-headless.\n",
            "(Reading database ... 122275 files and directories currently installed.)\n",
            "Preparing to unpack .../00-default-jre-headless_2%3a1.11-72build2_amd64.deb ...\n",
            "Unpacking default-jre-headless (2:1.11-72build2) ...\n",
            "Selecting previously unselected package libxtst6:amd64.\n",
            "Preparing to unpack .../01-libxtst6_2%3a1.2.3-1build4_amd64.deb ...\n",
            "Unpacking libxtst6:amd64 (2:1.2.3-1build4) ...\n",
            "Selecting previously unselected package openjdk-11-jre:amd64.\n",
            "Preparing to unpack .../02-openjdk-11-jre_11.0.23+9-1ubuntu1~22.04.1_amd64.deb ...\n",
            "Unpacking openjdk-11-jre:amd64 (11.0.23+9-1ubuntu1~22.04.1) ...\n",
            "Selecting previously unselected package default-jre.\n",
            "Preparing to unpack .../03-default-jre_2%3a1.11-72build2_amd64.deb ...\n",
            "Unpacking default-jre (2:1.11-72build2) ...\n",
            "Selecting previously unselected package libcommons-math3-java.\n",
            "Preparing to unpack .../04-libcommons-math3-java_3.6.1-3_all.deb ...\n",
            "Unpacking libcommons-math3-java (3.6.1-3) ...\n",
            "Selecting previously unselected package libjbzip2-java.\n",
            "Preparing to unpack .../05-libjbzip2-java_0.9.1-7_all.deb ...\n",
            "Unpacking libjbzip2-java (0.9.1-7) ...\n",
            "Selecting previously unselected package libapache-pom-java.\n",
            "Preparing to unpack .../06-libapache-pom-java_18-1_all.deb ...\n",
            "Unpacking libapache-pom-java (18-1) ...\n",
            "Selecting previously unselected package libcommons-parent-java.\n",
            "Preparing to unpack .../07-libcommons-parent-java_43-1_all.deb ...\n",
            "Unpacking libcommons-parent-java (43-1) ...\n",
            "Selecting previously unselected package libcommons-compress-java.\n",
            "Preparing to unpack .../08-libcommons-compress-java_1.21-1_all.deb ...\n",
            "Unpacking libcommons-compress-java (1.21-1) ...\n",
            "Selecting previously unselected package libcommons-logging-java.\n",
            "Preparing to unpack .../09-libcommons-logging-java_1.2-2_all.deb ...\n",
            "Unpacking libcommons-logging-java (1.2-2) ...\n",
            "Selecting previously unselected package libcommons-jexl2-java.\n",
            "Preparing to unpack .../10-libcommons-jexl2-java_2.1.1-5_all.deb ...\n",
            "Unpacking libcommons-jexl2-java (2.1.1-5) ...\n",
            "Selecting previously unselected package libjson-simple-java.\n",
            "Preparing to unpack .../11-libjson-simple-java_2.3.0-1_all.deb ...\n",
            "Unpacking libjson-simple-java (2.3.0-1) ...\n",
            "Selecting previously unselected package libngs-sdk2:amd64.\n",
            "Preparing to unpack .../12-libngs-sdk2_2.11.2-3_amd64.deb ...\n",
            "Unpacking libngs-sdk2:amd64 (2.11.2-3) ...\n",
            "Selecting previously unselected package libngs-sdk-dev:amd64.\n",
            "Preparing to unpack .../13-libngs-sdk-dev_2.11.2-3_amd64.deb ...\n",
            "Unpacking libngs-sdk-dev:amd64 (2.11.2-3) ...\n",
            "Selecting previously unselected package libngs-java:amd64.\n",
            "Preparing to unpack .../14-libngs-java_2.11.2-3_amd64.deb ...\n",
            "Unpacking libngs-java:amd64 (2.11.2-3) ...\n",
            "Selecting previously unselected package libsnappy-jni.\n",
            "Preparing to unpack .../15-libsnappy-jni_1.1.8.3-1_amd64.deb ...\n",
            "Unpacking libsnappy-jni (1.1.8.3-1) ...\n",
            "Selecting previously unselected package libsnappy-java.\n",
            "Preparing to unpack .../16-libsnappy-java_1.1.8.3-1_all.deb ...\n",
            "Unpacking libsnappy-java (1.1.8.3-1) ...\n",
            "Selecting previously unselected package libhtsjdk-java.\n",
            "Preparing to unpack .../17-libhtsjdk-java_2.24.1+dfsg-1_all.deb ...\n",
            "Unpacking libhtsjdk-java (2.24.1+dfsg-1) ...\n",
            "Selecting previously unselected package libfindbin-libs-perl.\n",
            "Preparing to unpack .../18-libfindbin-libs-perl_2.190.02-1_all.deb ...\n",
            "Unpacking libfindbin-libs-perl (2.190.02-1) ...\n",
            "Selecting previously unselected package libargs4j-java.\n",
            "Preparing to unpack .../19-libargs4j-java_2.33-1.1_all.deb ...\n",
            "Unpacking libargs4j-java (2.33-1.1) ...\n",
            "Selecting previously unselected package libcommons-io-java.\n",
            "Preparing to unpack .../20-libcommons-io-java_2.11.0-2_all.deb ...\n",
            "Unpacking libcommons-io-java (2.11.0-2) ...\n",
            "Selecting previously unselected package libcommons-lang3-java.\n",
            "Preparing to unpack .../21-libcommons-lang3-java_3.11-1_all.deb ...\n",
            "Unpacking libcommons-lang3-java (3.11-1) ...\n",
            "Selecting previously unselected package libsis-base-jni.\n",
            "Preparing to unpack .../22-libsis-base-jni_18.09~pre1+git20180928.45fbd31+dfsg-3_amd64.deb ...\n",
            "Unpacking libsis-base-jni (18.09~pre1+git20180928.45fbd31+dfsg-3) ...\n",
            "Selecting previously unselected package libsis-base-java.\n",
            "Preparing to unpack .../23-libsis-base-java_18.09~pre1+git20180928.45fbd31+dfsg-3_all.deb ...\n",
            "Unpacking libsis-base-java (18.09~pre1+git20180928.45fbd31+dfsg-3) ...\n",
            "Selecting previously unselected package libsis-jhdf5-jni.\n",
            "Preparing to unpack .../24-libsis-jhdf5-jni_19.04.0+dfsg-4_amd64.deb ...\n",
            "Unpacking libsis-jhdf5-jni (19.04.0+dfsg-4) ...\n",
            "Selecting previously unselected package libsis-jhdf5-java.\n",
            "Preparing to unpack .../25-libsis-jhdf5-java_19.04.0+dfsg-4_all.deb ...\n",
            "Unpacking libsis-jhdf5-java (19.04.0+dfsg-4) ...\n",
            "Selecting previously unselected package fastqc.\n",
            "Preparing to unpack .../26-fastqc_0.11.9+dfsg-5_all.deb ...\n",
            "Unpacking fastqc (0.11.9+dfsg-5) ...\n",
            "Selecting previously unselected package fonts-dejavu-core.\n",
            "Preparing to unpack .../27-fonts-dejavu-core_2.37-2build1_all.deb ...\n",
            "Unpacking fonts-dejavu-core (2.37-2build1) ...\n",
            "Selecting previously unselected package fonts-dejavu-extra.\n",
            "Preparing to unpack .../28-fonts-dejavu-extra_2.37-2build1_all.deb ...\n",
            "Unpacking fonts-dejavu-extra (2.37-2build1) ...\n",
            "Selecting previously unselected package libfontenc1:amd64.\n",
            "Preparing to unpack .../29-libfontenc1_1%3a1.1.4-1build3_amd64.deb ...\n",
            "Unpacking libfontenc1:amd64 (1:1.1.4-1build3) ...\n",
            "Selecting previously unselected package libxkbfile1:amd64.\n",
            "Preparing to unpack .../30-libxkbfile1_1%3a1.1.0-1build3_amd64.deb ...\n",
            "Unpacking libxkbfile1:amd64 (1:1.1.0-1build3) ...\n",
            "Selecting previously unselected package libxxf86dga1:amd64.\n",
            "Preparing to unpack .../31-libxxf86dga1_2%3a1.1.5-0ubuntu3_amd64.deb ...\n",
            "Unpacking libxxf86dga1:amd64 (2:1.1.5-0ubuntu3) ...\n",
            "Selecting previously unselected package x11-utils.\n",
            "Preparing to unpack .../32-x11-utils_7.7+5build2_amd64.deb ...\n",
            "Unpacking x11-utils (7.7+5build2) ...\n",
            "Selecting previously unselected package libatk-wrapper-java.\n",
            "Preparing to unpack .../33-libatk-wrapper-java_0.38.0-5build1_all.deb ...\n",
            "Unpacking libatk-wrapper-java (0.38.0-5build1) ...\n",
            "Selecting previously unselected package libatk-wrapper-java-jni:amd64.\n",
            "Preparing to unpack .../34-libatk-wrapper-java-jni_0.38.0-5build1_amd64.deb ...\n",
            "Unpacking libatk-wrapper-java-jni:amd64 (0.38.0-5build1) ...\n",
            "Setting up default-jre-headless (2:1.11-72build2) ...\n",
            "Setting up libxtst6:amd64 (2:1.2.3-1build4) ...\n",
            "Setting up libngs-sdk2:amd64 (2.11.2-3) ...\n",
            "Setting up libxxf86dga1:amd64 (2:1.1.5-0ubuntu3) ...\n",
            "Setting up openjdk-11-jre:amd64 (11.0.23+9-1ubuntu1~22.04.1) ...\n",
            "Setting up libfindbin-libs-perl (2.190.02-1) ...\n",
            "Setting up default-jre (2:1.11-72build2) ...\n",
            "Setting up libjson-simple-java (2.3.0-1) ...\n",
            "Setting up libsis-base-jni (18.09~pre1+git20180928.45fbd31+dfsg-3) ...\n",
            "Setting up libfontenc1:amd64 (1:1.1.4-1build3) ...\n",
            "Setting up libapache-pom-java (18-1) ...\n",
            "Setting up fonts-dejavu-core (2.37-2build1) ...\n",
            "Setting up libjbzip2-java (0.9.1-7) ...\n",
            "Setting up libngs-sdk-dev:amd64 (2.11.2-3) ...\n",
            "Setting up fonts-dejavu-extra (2.37-2build1) ...\n",
            "Setting up libcommons-math3-java (3.6.1-3) ...\n",
            "Setting up libsis-jhdf5-jni (19.04.0+dfsg-4) ...\n",
            "Setting up libxkbfile1:amd64 (1:1.1.0-1build3) ...\n",
            "Setting up libngs-java:amd64 (2.11.2-3) ...\n",
            "Setting up libsnappy-jni (1.1.8.3-1) ...\n",
            "Setting up libargs4j-java (2.33-1.1) ...\n",
            "Setting up libcommons-parent-java (43-1) ...\n",
            "Setting up libcommons-logging-java (1.2-2) ...\n",
            "Setting up libcommons-jexl2-java (2.1.1-5) ...\n",
            "Setting up libcommons-lang3-java (3.11-1) ...\n",
            "Setting up x11-utils (7.7+5build2) ...\n",
            "Setting up libsnappy-java (1.1.8.3-1) ...\n",
            "Setting up libatk-wrapper-java (0.38.0-5build1) ...\n",
            "Setting up libcommons-compress-java (1.21-1) ...\n",
            "Setting up libcommons-io-java (2.11.0-2) ...\n",
            "Setting up libatk-wrapper-java-jni:amd64 (0.38.0-5build1) ...\n",
            "Setting up libsis-base-java (18.09~pre1+git20180928.45fbd31+dfsg-3) ...\n",
            "Setting up libhtsjdk-java (2.24.1+dfsg-1) ...\n",
            "Setting up libsis-jhdf5-java (19.04.0+dfsg-4) ...\n",
            "Setting up fastqc (0.11.9+dfsg-5) ...\n",
            "Processing triggers for man-db (2.10.2-1) ...\n",
            "Processing triggers for menu (2.1.47ubuntu4) ...\n",
            "Processing triggers for fontconfig (2.13.1-4.2ubuntu5) ...\n",
            "Processing triggers for hicolor-icon-theme (0.17-2) ...\n",
            "Processing triggers for libc-bin (2.35-0ubuntu3.4) ...\n",
            "/sbin/ldconfig.real: /usr/local/lib/libtbbbind_2_0.so.3 is not a symbolic link\n",
            "\n",
            "/sbin/ldconfig.real: /usr/local/lib/libtbbmalloc_proxy.so.2 is not a symbolic link\n",
            "\n",
            "/sbin/ldconfig.real: /usr/local/lib/libtbbmalloc.so.2 is not a symbolic link\n",
            "\n",
            "/sbin/ldconfig.real: /usr/local/lib/libtbbbind.so.3 is not a symbolic link\n",
            "\n",
            "/sbin/ldconfig.real: /usr/local/lib/libtbb.so.12 is not a symbolic link\n",
            "\n",
            "/sbin/ldconfig.real: /usr/local/lib/libtbbbind_2_5.so.3 is not a symbolic link\n",
            "\n",
            "Collecting multiqc\n",
            "  Downloading multiqc-1.22.3-py3-none-any.whl (4.5 MB)\n",
            "\u001b[2K     \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m4.5/4.5 MB\u001b[0m \u001b[31m11.4 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hRequirement already satisfied: click in /usr/local/lib/python3.10/dist-packages (from multiqc) (8.1.7)\n",
            "Requirement already satisfied: humanize in /usr/local/lib/python3.10/dist-packages (from multiqc) (4.7.0)\n",
            "Requirement already satisfied: importlib-metadata in /usr/local/lib/python3.10/dist-packages (from multiqc) (7.1.0)\n",
            "Requirement already satisfied: jinja2>=3.0.0 in /usr/local/lib/python3.10/dist-packages (from multiqc) (3.1.4)\n",
            "Collecting kaleido (from multiqc)\n",
            "  Downloading kaleido-0.2.1-py2.py3-none-manylinux1_x86_64.whl (79.9 MB)\n",
            "\u001b[2K     \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m79.9/79.9 MB\u001b[0m \u001b[31m10.5 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hRequirement already satisfied: markdown in /usr/local/lib/python3.10/dist-packages (from multiqc) (3.6)\n",
            "Requirement already satisfied: numpy in /usr/local/lib/python3.10/dist-packages (from multiqc) (1.25.2)\n",
            "Requirement already satisfied: packaging in /usr/local/lib/python3.10/dist-packages (from multiqc) (24.1)\n",
            "Requirement already satisfied: requests in /usr/local/lib/python3.10/dist-packages (from multiqc) (2.31.0)\n",
            "Collecting Pillow>=10 (from multiqc)\n",
            "  Downloading pillow-10.3.0-cp310-cp310-manylinux_2_28_x86_64.whl (4.5 MB)\n",
            "\u001b[2K     \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m4.5/4.5 MB\u001b[0m \u001b[31m43.5 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hCollecting plotly>=5.18 (from multiqc)\n",
            "  Downloading plotly-5.22.0-py3-none-any.whl (16.4 MB)\n",
            "\u001b[2K     \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m16.4/16.4 MB\u001b[0m \u001b[31m73.3 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hRequirement already satisfied: pyyaml>=4 in /usr/local/lib/python3.10/dist-packages (from multiqc) (6.0.1)\n",
            "Collecting pyaml-env (from multiqc)\n",
            "  Downloading pyaml_env-1.2.1-py3-none-any.whl (9.0 kB)\n",
            "Requirement already satisfied: rich>=10 in /usr/local/lib/python3.10/dist-packages (from multiqc) (13.7.1)\n",
            "Collecting rich-click (from multiqc)\n",
            "  Downloading rich_click-1.8.3-py3-none-any.whl (35 kB)\n",
            "Collecting coloredlogs (from multiqc)\n",
            "  Downloading coloredlogs-15.0.1-py2.py3-none-any.whl (46 kB)\n",
            "\u001b[2K     \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m46.0/46.0 kB\u001b[0m \u001b[31m4.8 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hRequirement already satisfied: tqdm in /usr/local/lib/python3.10/dist-packages (from multiqc) (4.66.4)\n",
            "Collecting spectra>=0.0.10 (from multiqc)\n",
            "  Downloading spectra-0.0.11.tar.gz (18 kB)\n",
            "  Preparing metadata (setup.py) ... \u001b[?25l\u001b[?25hdone\n",
            "Requirement already satisfied: pydantic>=2.7.1 in /usr/local/lib/python3.10/dist-packages (from multiqc) (2.7.4)\n",
            "Collecting typeguard (from multiqc)\n",
            "  Downloading typeguard-4.3.0-py3-none-any.whl (35 kB)\n",
            "Requirement already satisfied: MarkupSafe>=2.0 in /usr/local/lib/python3.10/dist-packages (from jinja2>=3.0.0->multiqc) (2.1.5)\n",
            "Requirement already satisfied: tenacity>=6.2.0 in /usr/local/lib/python3.10/dist-packages (from plotly>=5.18->multiqc) (8.4.1)\n",
            "Requirement already satisfied: annotated-types>=0.4.0 in /usr/local/lib/python3.10/dist-packages (from pydantic>=2.7.1->multiqc) (0.7.0)\n",
            "Requirement already satisfied: pydantic-core==2.18.4 in /usr/local/lib/python3.10/dist-packages (from pydantic>=2.7.1->multiqc) (2.18.4)\n",
            "Requirement already satisfied: typing-extensions>=4.6.1 in /usr/local/lib/python3.10/dist-packages (from pydantic>=2.7.1->multiqc) (4.12.2)\n",
            "Requirement already satisfied: markdown-it-py>=2.2.0 in /usr/local/lib/python3.10/dist-packages (from rich>=10->multiqc) (3.0.0)\n",
            "Requirement already satisfied: pygments<3.0.0,>=2.13.0 in /usr/local/lib/python3.10/dist-packages (from rich>=10->multiqc) (2.16.1)\n",
            "Collecting colormath>=3.0.0 (from spectra>=0.0.10->multiqc)\n",
            "  Downloading colormath-3.0.0.tar.gz (39 kB)\n",
            "  Preparing metadata (setup.py) ... \u001b[?25l\u001b[?25hdone\n",
            "Collecting humanfriendly>=9.1 (from coloredlogs->multiqc)\n",
            "  Downloading humanfriendly-10.0-py2.py3-none-any.whl (86 kB)\n",
            "\u001b[2K     \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m86.8/86.8 kB\u001b[0m \u001b[31m10.8 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hRequirement already satisfied: zipp>=0.5 in /usr/local/lib/python3.10/dist-packages (from importlib-metadata->multiqc) (3.19.2)\n",
            "Requirement already satisfied: charset-normalizer<4,>=2 in /usr/local/lib/python3.10/dist-packages (from requests->multiqc) (3.3.2)\n",
            "Requirement already satisfied: idna<4,>=2.5 in /usr/local/lib/python3.10/dist-packages (from requests->multiqc) (3.7)\n",
            "Requirement already satisfied: urllib3<3,>=1.21.1 in /usr/local/lib/python3.10/dist-packages (from requests->multiqc) (2.0.7)\n",
            "Requirement already satisfied: certifi>=2017.4.17 in /usr/local/lib/python3.10/dist-packages (from requests->multiqc) (2024.6.2)\n",
            "Requirement already satisfied: networkx>=2.0 in /usr/local/lib/python3.10/dist-packages (from colormath>=3.0.0->spectra>=0.0.10->multiqc) (3.3)\n",
            "Requirement already satisfied: mdurl~=0.1 in /usr/local/lib/python3.10/dist-packages (from markdown-it-py>=2.2.0->rich>=10->multiqc) (0.1.2)\n",
            "Building wheels for collected packages: spectra, colormath\n",
            "  Building wheel for spectra (setup.py) ... \u001b[?25l\u001b[?25hdone\n",
            "  Created wheel for spectra: filename=spectra-0.0.11-py3-none-any.whl size=17467 sha256=cc0855c1bae5fcdf89858d3ffe9b7663d32b83a8e2340034c30180a06db0f50c\n",
            "  Stored in directory: /root/.cache/pip/wheels/bc/27/ec/cddaa9489a49ddcf702e1f4dc1791db28a3b0121e1ff254f97\n",
            "  Building wheel for colormath (setup.py) ... \u001b[?25l\u001b[?25hdone\n",
            "  Created wheel for colormath: filename=colormath-3.0.0-py3-none-any.whl size=39407 sha256=e82d7ed0132f65a726e9fc79b35e1c1687705c005da8143760d488f30f583d66\n",
            "  Stored in directory: /root/.cache/pip/wheels/ab/b3/4d/c0738759c25a1df01958068f162cf2a9dc3ab1da8b972cfcfc\n",
            "Successfully built spectra colormath\n",
            "Installing collected packages: kaleido, typeguard, pyaml-env, plotly, Pillow, humanfriendly, colormath, spectra, coloredlogs, rich-click, multiqc\n",
            "  Attempting uninstall: plotly\n",
            "    Found existing installation: plotly 5.15.0\n",
            "    Uninstalling plotly-5.15.0:\n",
            "      Successfully uninstalled plotly-5.15.0\n",
            "  Attempting uninstall: Pillow\n",
            "    Found existing installation: Pillow 9.4.0\n",
            "    Uninstalling Pillow-9.4.0:\n",
            "      Successfully uninstalled Pillow-9.4.0\n",
            "\u001b[31mERROR: pip's dependency resolver does not currently take into account all the packages that are installed. This behaviour is the source of the following dependency conflicts.\n",
            "imageio 2.31.6 requires pillow<10.1.0,>=8.3.2, but you have pillow 10.3.0 which is incompatible.\u001b[0m\u001b[31m\n",
            "\u001b[0mSuccessfully installed Pillow-10.3.0 coloredlogs-15.0.1 colormath-3.0.0 humanfriendly-10.0 kaleido-0.2.1 multiqc-1.22.3 plotly-5.22.0 pyaml-env-1.2.1 rich-click-1.8.3 spectra-0.0.11 typeguard-4.3.0\n"
          ]
        },
        {
          "output_type": "display_data",
          "data": {
            "application/vnd.colab-display-data+json": {
              "pip_warning": {
                "packages": [
                  "PIL"
                ]
              },
              "id": "a60bfe7ceefc485e9eb2bcbd609bb5bb"
            }
          },
          "metadata": {}
        },
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Reading package lists... Done\n",
            "Building dependency tree... Done\n",
            "Reading state information... Done\n",
            "The following NEW packages will be installed:\n",
            "  seqtk\n",
            "0 upgraded, 1 newly installed, 0 to remove and 45 not upgraded.\n",
            "Need to get 30.2 kB of archives.\n",
            "After this operation, 85.0 kB of additional disk space will be used.\n",
            "Get:1 http://archive.ubuntu.com/ubuntu jammy/universe amd64 seqtk amd64 1.3-2 [30.2 kB]\n",
            "Fetched 30.2 kB in 1s (49.7 kB/s)\n",
            "Selecting previously unselected package seqtk.\n",
            "(Reading database ... 122874 files and directories currently installed.)\n",
            "Preparing to unpack .../archives/seqtk_1.3-2_amd64.deb ...\n",
            "Unpacking seqtk (1.3-2) ...\n",
            "Setting up seqtk (1.3-2) ...\n",
            "Processing triggers for man-db (2.10.2-1) ...\n",
            "Reading package lists... Done\n",
            "Building dependency tree... Done\n",
            "Reading state information... Done\n",
            "The following NEW packages will be installed:\n",
            "  fastp\n",
            "0 upgraded, 1 newly installed, 0 to remove and 45 not upgraded.\n",
            "Need to get 193 kB of archives.\n",
            "After this operation, 640 kB of additional disk space will be used.\n",
            "Get:1 http://archive.ubuntu.com/ubuntu jammy/universe amd64 fastp amd64 0.20.1+dfsg-1 [193 kB]\n",
            "Fetched 193 kB in 1s (219 kB/s)\n",
            "Selecting previously unselected package fastp.\n",
            "(Reading database ... 122885 files and directories currently installed.)\n",
            "Preparing to unpack .../fastp_0.20.1+dfsg-1_amd64.deb ...\n",
            "Unpacking fastp (0.20.1+dfsg-1) ...\n",
            "Setting up fastp (0.20.1+dfsg-1) ...\n",
            "Processing triggers for man-db (2.10.2-1) ...\n",
            "\n",
            "\"fasterq-dump\" version 2.11.3\n",
            "\n",
            "\n",
            "FastQC v0.11.9\n",
            "fastp 0.20.1\n"
          ]
        }
      ],
      "source": [
        "# Instalar SRA Toolkit\n",
        "!apt-get update\n",
        "!apt-get install -y sra-toolkit\n",
        "# Instalación de FastQC y MultiQC\n",
        "!apt-get install fastqc\n",
        "!pip install multiqc\n",
        "#Instalacion de seqtk\n",
        "!apt-get install -y seqtk\n",
        "# Install fastp\n",
        "!apt-get install -y fastp\n",
        "\n",
        "# Verificar las instalaciones\n",
        "!fasterq-dump --version\n",
        "!seqtk 2>&1 | head -n 1\n",
        "!fastqc --version\n",
        "!fastp --version"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "gZ-oXIgOPKPy"
      },
      "source": [
        "## Repositorios de Datos Bioinformáticos\n",
        "\n",
        "Existen numerosos repositorios de datos bioinformáticos que facilitan el almacenamiento, acceso y reutilización de datos generados en investigaciones científicas. Uno de los repositorios más importantes es el NCBI (National Center for Biotechnology Information), que proporciona acceso a una variedad de bases de datos y herramientas bioinformáticas. Entre estas bases de datos, destaca el Sequence Read Archive (SRA), que es una plataforma diseñada para almacenar secuencias de alto rendimiento generadas por diversas tecnologías de secuenciación.\n",
        "\n",
        "El SRA es un recurso fundamental para los investigadores que necesitan acceder a datos de secuencias crudas. Los datos depositados en el SRA incluyen lecturas de secuenciación, junto con información detallada sobre las condiciones del experimento, los protocolos utilizados y la metainformación asociada. Esta información se organiza en una estructura jerárquica que incluye **BioProjects**, **Assays** y **Experiments**. Un **BioProject** agrupa proyectos relacionados y proporciona un contexto para los datos generados. Los **Assays** describen las metodologías y técnicas empleadas, mientras que los **Experiments** detallan los procedimientos específicos y las condiciones experimentales.\n",
        "\n",
        "Para ilustrar estos conceptos, trabajaremos con datos específicos disponibles en el SRA y un BioProject en particular. Puedes acceder a la información del experimento utilizando el siguiente enlace del [SRA para SRR6853380](https://www.ncbi.nlm.nih.gov/sra/?term=SRR6853380) y obtener más detalles sobre el contexto del proyecto en el [BioProject PRJNA438545](https://www.ncbi.nlm.nih.gov/bioproject/PRJNA438545).\n",
        "\n",
        "Estos recursos son esenciales para la investigación bioinformática, ya que permiten la replicación de estudios, la validación de resultados y el desarrollo de nuevos análisis basados en datos existentes."
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 2,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "F1GxeJMGtCRJ",
        "outputId": "0476dcd6-d3e4-4fab-98a0-168c52c58301"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "spots read      : 6,900,587\n",
            "reads read      : 13,801,174\n",
            "reads written   : 13,801,174\n"
          ]
        }
      ],
      "source": [
        "# Crear directorio para los datos\n",
        "!mkdir -p data\n",
        "\n",
        "# Fasterq-dump es una herramienta del sra-toolkit especifica para descargar los archivos FastQ paired-end desde SRA\n",
        "# Con el ID de acceso SRA que se quiere descargar (SRR6853380)\n",
        "!fasterq-dump SRR6853380 --split-files --outdir data\n",
        "# Comprimir los archivos FastQ para ahorrar espacio (Es una buena practica, el espacio se acaba rapido)\n",
        "!gzip data/*.fastq\n",
        "###Paciencia, toma un rato"
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "### Análisis de los Archivos FastQ Originales\n",
        "\n",
        "En esta sección, comenzamos explorando los archivos FastQ originales descargados del SRA. Primero, mostramos el tamaño de estos archivos, lo que nos da una idea de la cantidad de datos generados en el experimento.\n",
        "\n",
        "Notamos que los archivos son bastante grandes, lo cual es típico de experimentos de secuenciación masiva como el shotgun sequencing. Este tipo de experimento genera una gran cantidad de lecturas cortas, lo que resulta en archivos de gran tamaño.\n",
        "\n",
        "A continuación, mostramos las primeras cuatro secuencias de cada archivo FastQ. Esto nos permite observar la estructura y formato de las lecturas.\n",
        "\n",
        "Al inspeccionar estas lecturas, podemos ver que cada lectura en el archivo FastQ consta de cuatro líneas:\n",
        "1. Una línea de encabezado que empieza con `@` seguida de un identificador único.\n",
        "2. La secuencia de nucleótidos.\n",
        "3. Una línea que empieza con `+` que puede repetir el identificador del encabezado.\n",
        "4. Una línea de calidad que corresponde a la secuencia de nucleótidos.\n",
        "\n",
        "Finalmente, contamos el número de lecturas en cada archivo FastQ. Esto es importante para cuantificar la cobertura y la profundidad del experimento.\n",
        "\n",
        "La cantidad de lecturas nos confirma que estamos trabajando con un experimento de secuenciación shotgun, caracterizado por generar millones de lecturas cortas para una cobertura exhaustiva del genoma o metagenoma de interés.\n"
      ],
      "metadata": {
        "id": "WEJmhvCU9_ES"
      }
    },
    {
      "cell_type": "code",
      "execution_count": 3,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "CJFjX3KYtrK0",
        "outputId": "4e72201e-e94c-42e9-8fef-5e501b0277ff"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Tamaño de los archivos FastQ originales:\n",
            "-rw-r--r-- 1 root root 468M Jun 24 02:02 data/SRR6853380_1.fastq.gz\n",
            "-rw-r--r-- 1 root root 475M Jun 24 02:02 data/SRR6853380_2.fastq.gz\n",
            "Lecturas Forward\n",
            "@SRR6853380.1 1 length=94\n",
            "GTATAATCGCACCTCCAAATAAAATTGAGGTGAAAAATGTCAAATCCCAAGCCGCTTTATATGGTGCTTGCTTATGATTCAAGTGCTCTGCCTA\n",
            "+SRR6853380.1 1 length=94\n",
            "BBBFFFFFF<FFFFFIFIIFIIIFFFB<FFBFIIIBFIFFFFBBFFF'7<7''7B77BFFFFBFFFFF<F'B<BF7'7<'<<'707<'<<B<BB\n",
            "@SRR6853380.2 2 length=40\n",
            "GTGCCGGGCATCTCCAGCAGCGGGCGCCACCCCCATTCGC\n",
            "+SRR6853380.2 2 length=40\n",
            "BBBFFFFFB<FFFFI7FFBBFFFF<F7'''7BF'77<700\n",
            "@SRR6853380.3 3 length=45\n",
            "GACCAGCGTCATCTCATGCAGCGCTTTGCCCGGCAGGTGCTTCCG\n",
            "+SRR6853380.3 3 length=45\n",
            "BBBFFFFFFF7BB<BB<BFF<BFFFFFFFFIIIIIII0<0''0BB\n",
            "@SRR6853380.4 4 length=101\n",
            "TTGCCACTGCGCCAGCGTAAAGAAGAGGTTGCTTTCGCCGCGCATCACGCGATAGGTGTTCTTGTCGAAGCTCACCACATAGCCGATCACGGCCACCATGC\n",
            "+SRR6853380.4 4 length=101\n",
            "BBBFFFFFFFFFFIIIIFFFFFIIIFFFFFFFFFIFBFBFBFIFFFFFFFBBFBFBBBBB<B0<BBFB70'<BBFBBBBB<<<B<BBBBBB<7<BBFBB<7\n",
            "Lecturas Reverse\n",
            "@SRR6853380.1 1 length=101\n",
            "TATCAAGACTTTGTCCAAATTAGCATCGACAGTTTCAATGTGGTCATACGCGTACAGTGCATGGTTTTTAGCTGAATGGATNNNNNNNNNNATTTCTTCGG\n",
            "+SRR6853380.1 1 length=101\n",
            "BBBFFBFBBBBFBFFFFFFFFBFBBBB<0BBFBFFBFFBBBBBB<BFFBFB<70<<B<BB0<BB<BBFBF7<<<''7<707##########'0000<B0<7\n",
            "@SRR6853380.2 2 length=60\n",
            "GCCCTGGCCTTTGCTGTAGAGCAGATAGTTCGGGTTGGTCTGGTAGAGACTCGGCGCTTT\n",
            "+SRR6853380.2 2 length=60\n",
            "<<BF<<BFF<FFBBBB<<B<<BFFBF<B0<BFFFF<FF7<BBBBF<0'''0<BB7<<'7B\n",
            "@SRR6853380.3 3 length=46\n",
            "ATGTATGACGGCATCCAGTATCGCGGCTAAGGCCCGGCGCTGGTGG\n",
            "+SRR6853380.3 3 length=46\n",
            "BBBBBBBBFFFB0BBFFFB7FBBFFF<B'07FBF0'<7<F077'7<\n",
            "@SRR6853380.4 4 length=55\n",
            "CGCTGGTTCCAACGTTATCGTCCGCTTTCGCTACGCGCCCGCTTTTTACTGGCCT\n",
            "+SRR6853380.4 4 length=55\n",
            "BBBFFFFFFFFFFBFFFIFBBFFFFIIIII7FFFFIBFIFFFFFFFFFFF<<BBB\n",
            "Número de lecturas en SRR6853380_1.fastq.gz:\n",
            "6900587\n",
            "Número de lecturas en SRR6853380_2.fastq.gz:\n",
            "6900587\n"
          ]
        }
      ],
      "source": [
        "# Mostrar el tamaño de los archivos FastQ originales\n",
        "!echo \"Tamaño de los archivos FastQ originales:\"\n",
        "!ls -lh data/*.fastq.gz\n",
        "\n",
        "# Mostrar las primeras 4 secuencias de cada archivo FastQ\n",
        "!echo \"Lecturas Forward\"\n",
        "!zcat data/SRR6853380_1.fastq.gz | head -n 16\n",
        "!echo \"Lecturas Reverse\"\n",
        "!zcat data/SRR6853380_2.fastq.gz | head -n 16\n",
        "\n",
        "# Contar el número de lecturas en cada archivo FastQ\n",
        "!echo \"Número de lecturas en SRR6853380_1.fastq.gz:\"\n",
        "!zcat data/SRR6853380_1.fastq.gz | echo $((`wc -l`/4))\n",
        "\n",
        "!echo \"Número de lecturas en SRR6853380_2.fastq.gz:\"\n",
        "!zcat data/SRR6853380_2.fastq.gz | echo $((`wc -l`/4))"
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "### Extracción de un Subconjunto de Lecturas con `seqtk`\n",
        "\n",
        "En algunas ocasiones, es necesario trabajar con un subconjunto de datos en lugar del conjunto completo, especialmente cuando se manejan archivos de gran tamaño. Esto permite realizar análisis preliminares más rápidos y menos intensivos en recursos. Para este propósito, utilizamos `seqtk`, una herramienta eficiente para manipular archivos FastQ.\n",
        "\n",
        "`seqtk` es una herramienta versátil que permite realizar diversas operaciones en archivos de secuencias, como filtrado, recorte y muestreo aleatorio. En este caso, utilizamos `seqtk` para extraer un subconjunto de 100,000 lecturas de los archivos FastQ originales.\n",
        "\n",
        "Después de extraer el subconjunto, comprimimos los archivos resultantes para ahorrar espacio en disco. A continuación, mostramos el tamaño de los archivos FastQ del subconjunto y contamos el número de lecturas en cada uno de ellos para asegurarnos de que la extracción se realizó correctamente.\n",
        "\n",
        "Este proceso de fragmentación de datos es útil en varias situaciones, como:\n",
        "- Realizar pruebas rápidas antes de ejecutar análisis completos.\n",
        "- Comparar diferentes métodos de análisis en un subconjunto representativo.\n",
        "- Manejar limitaciones de memoria y tiempo de procesamiento en equipos con recursos limitados."
      ],
      "metadata": {
        "id": "UzviQ2-gA-FI"
      }
    },
    {
      "cell_type": "code",
      "execution_count": 4,
      "metadata": {
        "id": "m_detVmG2d75",
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "47ba019d-fcb4-4dbf-c96a-e4e443740ae2"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Tamaño de los archivos FastQ del subconjunto:\n",
            "-rw-r--r-- 1 root root 6.6M Jun 24 02:18 data/subset_SRR6853380_1.fastq.gz\n",
            "-rw-r--r-- 1 root root 6.7M Jun 24 02:18 data/subset_SRR6853380_2.fastq.gz\n",
            "Número de lecturas en SRR6853380_1.fastq.gz:\n",
            "100000\n",
            "Número de lecturas en SRR6853380_2.fastq.gz:\n",
            "100000\n"
          ]
        }
      ],
      "source": [
        "# Número de lecturas a extraer para el subconjunto\n",
        "num_reads = 100000\n",
        "\n",
        "# Extraer un subconjunto del archivo FastQ usando seqtk\n",
        "!seqtk sample -s100 data/SRR6853380_1.fastq.gz $num_reads > data/subset_SRR6853380_1.fastq\n",
        "!seqtk sample -s100 data/SRR6853380_2.fastq.gz $num_reads > data/subset_SRR6853380_2.fastq\n",
        "\n",
        "# Comprimir el subconjunto para ahorrar espacio\n",
        "!gzip data/subset_SRR6853380_1.fastq\n",
        "!gzip data/subset_SRR6853380_2.fastq\n",
        "\n",
        "# Mostrar el tamaño de los archivos FastQ del subconjunto\n",
        "!echo \"Tamaño de los archivos FastQ del subconjunto:\"\n",
        "!ls -lh data/subset_SRR6853380_*.fastq.gz\n",
        "\n",
        "# Contar el número de lecturas en cada archivo FastQ\n",
        "!echo \"Número de lecturas en SRR6853380_1.fastq.gz:\"\n",
        "!zcat data/subset_SRR6853380_1.fastq.gz | echo $((`wc -l`/4))\n",
        "\n",
        "!echo \"Número de lecturas en SRR6853380_2.fastq.gz:\"\n",
        "!zcat data/subset_SRR6853380_2.fastq.gz | echo $((`wc -l`/4))\n"
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "### Trimming de los Datos FastQ con `fastp`\n",
        "\n",
        "El trimming es un paso crucial en el preprocesamiento de datos de secuenciación. Consiste en eliminar bases de baja calidad y adaptadores de secuencias, lo cual mejora la precisión y fiabilidad de los análisis posteriores. Esto es especialmente importante en experimentos de secuenciación masiva donde la calidad de las lecturas puede variar significativamente.\n",
        "\n",
        "Para realizar el trimming, utilizamos `fastp`, una herramienta rápida y versátil que no solo realiza el trimming sino que también genera informes detallados sobre la calidad de los datos antes y después del proceso.\n",
        "\n",
        "En este caso, aplicamos el trimming tanto a los datos completos como a los subconjuntos previamente generados. Esto nos permitirá comparar los efectos del trimming en ambos conjuntos de datos utilizando FastQC en análisis posteriores.\n",
        "\n",
        "El trimming ayuda a:\n",
        "- Mejorar la calidad general de las lecturas.\n",
        "- Reducir el ruido en los datos, lo que puede interferir con los análisis bioinformáticos.\n",
        "- Aumentar la precisión de la alineación y el ensamblaje de secuencias.\n",
        "\n",
        "Ejecutamos `fastp` para recortar las lecturas en los archivos FastQ y guardamos los resultados en nuevos archivos comprimidos."
      ],
      "metadata": {
        "id": "0Vd_P2gTBC8B"
      }
    },
    {
      "cell_type": "code",
      "execution_count": 5,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "niTzgS5V4_Vk",
        "outputId": "9e39869d-aa90-495f-e537-afae6b359636"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Read1 before filtering:\n",
            "total reads: 100000\n",
            "total bases: 9166546\n",
            "Q20 bases: 8212105(89.5878%)\n",
            "Q30 bases: 6805338(74.241%)\n",
            "\n",
            "Read2 before filtering:\n",
            "total reads: 100000\n",
            "total bases: 9157497\n",
            "Q20 bases: 8082548(88.2615%)\n",
            "Q30 bases: 6558610(71.6201%)\n",
            "\n",
            "Read1 after filtering:\n",
            "total reads: 98063\n",
            "total bases: 8994921\n",
            "Q20 bases: 8087550(89.9124%)\n",
            "Q30 bases: 6705952(74.5527%)\n",
            "\n",
            "Read2 aftering filtering:\n",
            "total reads: 98063\n",
            "total bases: 8975429\n",
            "Q20 bases: 8002200(89.1567%)\n",
            "Q30 bases: 6512419(72.5583%)\n",
            "\n",
            "Filtering result:\n",
            "reads passed filter: 196126\n",
            "reads failed due to low quality: 3476\n",
            "reads failed due to too many N: 398\n",
            "reads failed due to too short: 0\n",
            "reads with adapter trimmed: 118\n",
            "bases trimmed due to adapters: 1145\n",
            "\n",
            "Duplication rate: 0.792252%\n",
            "\n",
            "Insert size peak (evaluated by paired-end reads): 141\n",
            "\n",
            "JSON report: fastp.json\n",
            "HTML report: fastp.html\n",
            "\n",
            "fastp -i data/subset_SRR6853380_1.fastq.gz -I data/subset_SRR6853380_2.fastq.gz -o data/subset_trimmed_SRR6853380_1.fastq.gz -O data/subset_trimmed_SRR6853380_2.fastq.gz \n",
            "fastp v0.20.1, time used: 3 seconds\n",
            "Read1 before filtering:\n",
            "total reads: 6900587\n",
            "total bases: 632359080\n",
            "Q20 bases: 566693186(89.6157%)\n",
            "Q30 bases: 469784510(74.2908%)\n",
            "\n",
            "Read2 before filtering:\n",
            "total reads: 6900587\n",
            "total bases: 632234950\n",
            "Q20 bases: 558037303(88.2642%)\n",
            "Q30 bases: 452768674(71.614%)\n",
            "\n",
            "Read1 after filtering:\n",
            "total reads: 6764982\n",
            "total bases: 620303838\n",
            "Q20 bases: 557934036(89.9453%)\n",
            "Q30 bases: 462827327(74.613%)\n",
            "\n",
            "Read2 aftering filtering:\n",
            "total reads: 6764982\n",
            "total bases: 619424875\n",
            "Q20 bases: 552416078(89.1821%)\n",
            "Q30 bases: 449564371(72.5777%)\n",
            "\n",
            "Filtering result:\n",
            "reads passed filter: 13529964\n",
            "reads failed due to low quality: 241674\n",
            "reads failed due to too many N: 29536\n",
            "reads failed due to too short: 0\n",
            "reads with adapter trimmed: 6460\n",
            "bases trimmed due to adapters: 82180\n",
            "\n",
            "Duplication rate: 18.0164%\n",
            "\n",
            "Insert size peak (evaluated by paired-end reads): 166\n",
            "\n",
            "JSON report: fastp.json\n",
            "HTML report: fastp.html\n",
            "\n",
            "fastp -i data/SRR6853380_1.fastq.gz -I data/SRR6853380_2.fastq.gz -o data/trimmed_SRR6853380_1.fastq.gz -O data/trimmed_SRR6853380_2.fastq.gz \n",
            "fastp v0.20.1, time used: 205 seconds\n"
          ]
        }
      ],
      "source": [
        "# Trim the FastQ data using fastp\n",
        "!fastp -i data/subset_SRR6853380_1.fastq.gz -I data/subset_SRR6853380_2.fastq.gz -o data/subset_trimmed_SRR6853380_1.fastq.gz -O data/subset_trimmed_SRR6853380_2.fastq.gz\n",
        "!fastp -i data/SRR6853380_1.fastq.gz -I data/SRR6853380_2.fastq.gz -o data/trimmed_SRR6853380_1.fastq.gz -O data/trimmed_SRR6853380_2.fastq.gz"
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "### Análisis de Calidad con FastQC y MultiQC\n",
        "\n",
        "Para evaluar la calidad de nuestros datos de secuenciación, utilizamos dos herramientas clave: FastQC y MultiQC.\n",
        "\n",
        "**FastQC** es una herramienta ampliamente utilizada para realizar un control de calidad rápido de los datos de secuenciación. Genera una serie de gráficos e informes que nos permiten evaluar varios aspectos de la calidad de las lecturas, como la distribución de la calidad de las bases, el contenido GC, la presencia de adaptadores, entre otros.\n",
        "\n",
        "**MultiQC** es una herramienta que agrega los resultados de múltiples ejecuciones de FastQC (u otras herramientas de análisis de calidad) en un único informe comprensivo. Esto es particularmente útil cuando se trabaja con muchos archivos FastQ, ya que permite una visualización y comparación centralizada de todos los datos de calidad.\n",
        "\n",
        "En esta sección, ejecutamos FastQC en todas las muestras generadas hasta ahora (incluyendo los datos originales, los subconjuntos y los datos recortados). Luego, utilizamos MultiQC para generar un informe combinado que nos permita comparar fácilmente la calidad de todos estos conjuntos de datos.\n",
        "\n",
        "Para una descripción detallada de los informes generados por FastQC y qué esperar de cada métrica dependiendo del tipo de experimento, se recomienda revisar la [documentación oficial de MultiQC](https://multiqc.info/), que ofrece un desglosado exhaustivo de los informes de calidad.\n"
      ],
      "metadata": {
        "id": "ROtBwBrnGp5Y"
      }
    },
    {
      "cell_type": "code",
      "execution_count": 6,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "zI1-4ugH-N2u",
        "outputId": "8643c1d1-252a-4f25-afaf-8fa11e00d78c"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Started analysis of SRR6853380_1.fastq.gz\n",
            "Approx 5% complete for SRR6853380_1.fastq.gz\n",
            "Approx 10% complete for SRR6853380_1.fastq.gz\n",
            "Approx 15% complete for SRR6853380_1.fastq.gz\n",
            "Approx 20% complete for SRR6853380_1.fastq.gz\n",
            "Approx 25% complete for SRR6853380_1.fastq.gz\n",
            "Approx 30% complete for SRR6853380_1.fastq.gz\n",
            "Approx 35% complete for SRR6853380_1.fastq.gz\n",
            "Approx 40% complete for SRR6853380_1.fastq.gz\n",
            "Approx 45% complete for SRR6853380_1.fastq.gz\n",
            "Approx 50% complete for SRR6853380_1.fastq.gz\n",
            "Approx 55% complete for SRR6853380_1.fastq.gz\n",
            "Approx 60% complete for SRR6853380_1.fastq.gz\n",
            "Approx 65% complete for SRR6853380_1.fastq.gz\n",
            "Approx 70% complete for SRR6853380_1.fastq.gz\n",
            "Approx 75% complete for SRR6853380_1.fastq.gz\n",
            "Approx 80% complete for SRR6853380_1.fastq.gz\n",
            "Approx 85% complete for SRR6853380_1.fastq.gz\n",
            "Approx 90% complete for SRR6853380_1.fastq.gz\n",
            "Approx 95% complete for SRR6853380_1.fastq.gz\n",
            "Analysis complete for SRR6853380_1.fastq.gz\n",
            "Started analysis of SRR6853380_2.fastq.gz\n",
            "Approx 5% complete for SRR6853380_2.fastq.gz\n",
            "Approx 10% complete for SRR6853380_2.fastq.gz\n",
            "Approx 15% complete for SRR6853380_2.fastq.gz\n",
            "Approx 20% complete for SRR6853380_2.fastq.gz\n",
            "Approx 25% complete for SRR6853380_2.fastq.gz\n",
            "Approx 30% complete for SRR6853380_2.fastq.gz\n",
            "Approx 35% complete for SRR6853380_2.fastq.gz\n",
            "Approx 40% complete for SRR6853380_2.fastq.gz\n",
            "Approx 45% complete for SRR6853380_2.fastq.gz\n",
            "Approx 50% complete for SRR6853380_2.fastq.gz\n",
            "Approx 55% complete for SRR6853380_2.fastq.gz\n",
            "Approx 60% complete for SRR6853380_2.fastq.gz\n",
            "Approx 65% complete for SRR6853380_2.fastq.gz\n",
            "Approx 70% complete for SRR6853380_2.fastq.gz\n",
            "Approx 75% complete for SRR6853380_2.fastq.gz\n",
            "Approx 80% complete for SRR6853380_2.fastq.gz\n",
            "Approx 85% complete for SRR6853380_2.fastq.gz\n",
            "Approx 90% complete for SRR6853380_2.fastq.gz\n",
            "Approx 95% complete for SRR6853380_2.fastq.gz\n",
            "Analysis complete for SRR6853380_2.fastq.gz\n",
            "Started analysis of subset_SRR6853380_1.fastq.gz\n",
            "Approx 5% complete for subset_SRR6853380_1.fastq.gz\n",
            "Approx 10% complete for subset_SRR6853380_1.fastq.gz\n",
            "Approx 15% complete for subset_SRR6853380_1.fastq.gz\n",
            "Approx 20% complete for subset_SRR6853380_1.fastq.gz\n",
            "Approx 25% complete for subset_SRR6853380_1.fastq.gz\n",
            "Approx 30% complete for subset_SRR6853380_1.fastq.gz\n",
            "Approx 35% complete for subset_SRR6853380_1.fastq.gz\n",
            "Approx 40% complete for subset_SRR6853380_1.fastq.gz\n",
            "Approx 45% complete for subset_SRR6853380_1.fastq.gz\n",
            "Approx 50% complete for subset_SRR6853380_1.fastq.gz\n",
            "Approx 55% complete for subset_SRR6853380_1.fastq.gz\n",
            "Approx 60% complete for subset_SRR6853380_1.fastq.gz\n",
            "Approx 65% complete for subset_SRR6853380_1.fastq.gz\n",
            "Approx 70% complete for subset_SRR6853380_1.fastq.gz\n",
            "Approx 75% complete for subset_SRR6853380_1.fastq.gz\n",
            "Approx 80% complete for subset_SRR6853380_1.fastq.gz\n",
            "Approx 85% complete for subset_SRR6853380_1.fastq.gz\n",
            "Approx 90% complete for subset_SRR6853380_1.fastq.gz\n",
            "Approx 95% complete for subset_SRR6853380_1.fastq.gz\n",
            "Approx 100% complete for subset_SRR6853380_1.fastq.gz\n",
            "Analysis complete for subset_SRR6853380_1.fastq.gz\n",
            "Started analysis of subset_SRR6853380_2.fastq.gz\n",
            "Approx 5% complete for subset_SRR6853380_2.fastq.gz\n",
            "Approx 10% complete for subset_SRR6853380_2.fastq.gz\n",
            "Approx 15% complete for subset_SRR6853380_2.fastq.gz\n",
            "Approx 20% complete for subset_SRR6853380_2.fastq.gz\n",
            "Approx 25% complete for subset_SRR6853380_2.fastq.gz\n",
            "Approx 30% complete for subset_SRR6853380_2.fastq.gz\n",
            "Approx 35% complete for subset_SRR6853380_2.fastq.gz\n",
            "Approx 40% complete for subset_SRR6853380_2.fastq.gz\n",
            "Approx 45% complete for subset_SRR6853380_2.fastq.gz\n",
            "Approx 50% complete for subset_SRR6853380_2.fastq.gz\n",
            "Approx 55% complete for subset_SRR6853380_2.fastq.gz\n",
            "Approx 60% complete for subset_SRR6853380_2.fastq.gz\n",
            "Approx 65% complete for subset_SRR6853380_2.fastq.gz\n",
            "Approx 70% complete for subset_SRR6853380_2.fastq.gz\n",
            "Approx 75% complete for subset_SRR6853380_2.fastq.gz\n",
            "Approx 80% complete for subset_SRR6853380_2.fastq.gz\n",
            "Approx 85% complete for subset_SRR6853380_2.fastq.gz\n",
            "Approx 90% complete for subset_SRR6853380_2.fastq.gz\n",
            "Approx 95% complete for subset_SRR6853380_2.fastq.gz\n",
            "Approx 100% complete for subset_SRR6853380_2.fastq.gz\n",
            "Analysis complete for subset_SRR6853380_2.fastq.gz\n",
            "Started analysis of subset_trimmed_SRR6853380_1.fastq.gz\n",
            "Approx 5% complete for subset_trimmed_SRR6853380_1.fastq.gz\n",
            "Approx 10% complete for subset_trimmed_SRR6853380_1.fastq.gz\n",
            "Approx 15% complete for subset_trimmed_SRR6853380_1.fastq.gz\n",
            "Approx 20% complete for subset_trimmed_SRR6853380_1.fastq.gz\n",
            "Approx 25% complete for subset_trimmed_SRR6853380_1.fastq.gz\n",
            "Approx 30% complete for subset_trimmed_SRR6853380_1.fastq.gz\n",
            "Approx 35% complete for subset_trimmed_SRR6853380_1.fastq.gz\n",
            "Approx 40% complete for subset_trimmed_SRR6853380_1.fastq.gz\n",
            "Approx 45% complete for subset_trimmed_SRR6853380_1.fastq.gz\n",
            "Approx 50% complete for subset_trimmed_SRR6853380_1.fastq.gz\n",
            "Approx 55% complete for subset_trimmed_SRR6853380_1.fastq.gz\n",
            "Approx 60% complete for subset_trimmed_SRR6853380_1.fastq.gz\n",
            "Approx 65% complete for subset_trimmed_SRR6853380_1.fastq.gz\n",
            "Approx 70% complete for subset_trimmed_SRR6853380_1.fastq.gz\n",
            "Approx 75% complete for subset_trimmed_SRR6853380_1.fastq.gz\n",
            "Approx 80% complete for subset_trimmed_SRR6853380_1.fastq.gz\n",
            "Approx 85% complete for subset_trimmed_SRR6853380_1.fastq.gz\n",
            "Approx 90% complete for subset_trimmed_SRR6853380_1.fastq.gz\n",
            "Approx 95% complete for subset_trimmed_SRR6853380_1.fastq.gz\n",
            "Analysis complete for subset_trimmed_SRR6853380_1.fastq.gz\n",
            "Started analysis of subset_trimmed_SRR6853380_2.fastq.gz\n",
            "Approx 5% complete for subset_trimmed_SRR6853380_2.fastq.gz\n",
            "Approx 10% complete for subset_trimmed_SRR6853380_2.fastq.gz\n",
            "Approx 15% complete for subset_trimmed_SRR6853380_2.fastq.gz\n",
            "Approx 20% complete for subset_trimmed_SRR6853380_2.fastq.gz\n",
            "Approx 25% complete for subset_trimmed_SRR6853380_2.fastq.gz\n",
            "Approx 30% complete for subset_trimmed_SRR6853380_2.fastq.gz\n",
            "Approx 35% complete for subset_trimmed_SRR6853380_2.fastq.gz\n",
            "Approx 40% complete for subset_trimmed_SRR6853380_2.fastq.gz\n",
            "Approx 45% complete for subset_trimmed_SRR6853380_2.fastq.gz\n",
            "Approx 50% complete for subset_trimmed_SRR6853380_2.fastq.gz\n",
            "Approx 55% complete for subset_trimmed_SRR6853380_2.fastq.gz\n",
            "Approx 60% complete for subset_trimmed_SRR6853380_2.fastq.gz\n",
            "Approx 65% complete for subset_trimmed_SRR6853380_2.fastq.gz\n",
            "Approx 70% complete for subset_trimmed_SRR6853380_2.fastq.gz\n",
            "Approx 75% complete for subset_trimmed_SRR6853380_2.fastq.gz\n",
            "Approx 80% complete for subset_trimmed_SRR6853380_2.fastq.gz\n",
            "Approx 85% complete for subset_trimmed_SRR6853380_2.fastq.gz\n",
            "Approx 90% complete for subset_trimmed_SRR6853380_2.fastq.gz\n",
            "Approx 95% complete for subset_trimmed_SRR6853380_2.fastq.gz\n",
            "Analysis complete for subset_trimmed_SRR6853380_2.fastq.gz\n",
            "Started analysis of trimmed_SRR6853380_1.fastq.gz\n",
            "Approx 5% complete for trimmed_SRR6853380_1.fastq.gz\n",
            "Approx 10% complete for trimmed_SRR6853380_1.fastq.gz\n",
            "Approx 15% complete for trimmed_SRR6853380_1.fastq.gz\n",
            "Approx 20% complete for trimmed_SRR6853380_1.fastq.gz\n",
            "Approx 25% complete for trimmed_SRR6853380_1.fastq.gz\n",
            "Approx 30% complete for trimmed_SRR6853380_1.fastq.gz\n",
            "Approx 35% complete for trimmed_SRR6853380_1.fastq.gz\n",
            "Approx 40% complete for trimmed_SRR6853380_1.fastq.gz\n",
            "Approx 45% complete for trimmed_SRR6853380_1.fastq.gz\n",
            "Approx 50% complete for trimmed_SRR6853380_1.fastq.gz\n",
            "Approx 55% complete for trimmed_SRR6853380_1.fastq.gz\n",
            "Approx 60% complete for trimmed_SRR6853380_1.fastq.gz\n",
            "Approx 65% complete for trimmed_SRR6853380_1.fastq.gz\n",
            "Approx 70% complete for trimmed_SRR6853380_1.fastq.gz\n",
            "Approx 75% complete for trimmed_SRR6853380_1.fastq.gz\n",
            "Approx 80% complete for trimmed_SRR6853380_1.fastq.gz\n",
            "Approx 85% complete for trimmed_SRR6853380_1.fastq.gz\n",
            "Approx 90% complete for trimmed_SRR6853380_1.fastq.gz\n",
            "Approx 95% complete for trimmed_SRR6853380_1.fastq.gz\n",
            "Analysis complete for trimmed_SRR6853380_1.fastq.gz\n",
            "Started analysis of trimmed_SRR6853380_2.fastq.gz\n",
            "Approx 5% complete for trimmed_SRR6853380_2.fastq.gz\n",
            "Approx 10% complete for trimmed_SRR6853380_2.fastq.gz\n",
            "Approx 15% complete for trimmed_SRR6853380_2.fastq.gz\n",
            "Approx 20% complete for trimmed_SRR6853380_2.fastq.gz\n",
            "Approx 25% complete for trimmed_SRR6853380_2.fastq.gz\n",
            "Approx 30% complete for trimmed_SRR6853380_2.fastq.gz\n",
            "Approx 35% complete for trimmed_SRR6853380_2.fastq.gz\n",
            "Approx 40% complete for trimmed_SRR6853380_2.fastq.gz\n",
            "Approx 45% complete for trimmed_SRR6853380_2.fastq.gz\n",
            "Approx 50% complete for trimmed_SRR6853380_2.fastq.gz\n",
            "Approx 55% complete for trimmed_SRR6853380_2.fastq.gz\n",
            "Approx 60% complete for trimmed_SRR6853380_2.fastq.gz\n",
            "Approx 65% complete for trimmed_SRR6853380_2.fastq.gz\n",
            "Approx 70% complete for trimmed_SRR6853380_2.fastq.gz\n",
            "Approx 75% complete for trimmed_SRR6853380_2.fastq.gz\n",
            "Approx 80% complete for trimmed_SRR6853380_2.fastq.gz\n",
            "Approx 85% complete for trimmed_SRR6853380_2.fastq.gz\n",
            "Approx 90% complete for trimmed_SRR6853380_2.fastq.gz\n",
            "Approx 95% complete for trimmed_SRR6853380_2.fastq.gz\n",
            "Analysis complete for trimmed_SRR6853380_2.fastq.gz\n",
            "SRR6853380_1_fastqc.html\t subset_trimmed_SRR6853380_1_fastqc.html\n",
            "SRR6853380_1_fastqc.zip\t\t subset_trimmed_SRR6853380_1_fastqc.zip\n",
            "SRR6853380_2_fastqc.html\t subset_trimmed_SRR6853380_2_fastqc.html\n",
            "SRR6853380_2_fastqc.zip\t\t subset_trimmed_SRR6853380_2_fastqc.zip\n",
            "subset_SRR6853380_1_fastqc.html  trimmed_SRR6853380_1_fastqc.html\n",
            "subset_SRR6853380_1_fastqc.zip\t trimmed_SRR6853380_1_fastqc.zip\n",
            "subset_SRR6853380_2_fastqc.html  trimmed_SRR6853380_2_fastqc.html\n",
            "subset_SRR6853380_2_fastqc.zip\t trimmed_SRR6853380_2_fastqc.zip\n",
            "\n",
            "\u001b[91m///\u001b[0m \u001b]8;id=839551;https://multiqc.info\u001b\\\u001b[1mMultiQC\u001b[0m\u001b]8;;\u001b\\ 🔍 \u001b[2mv1.22.3\u001b[0m\n",
            "\n",
            "\u001b[34m       file_search\u001b[0m | Search path: /content/QC_results\n",
            "\u001b[2K         \u001b[34msearching\u001b[0m | \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[35m100%\u001b[0m \u001b[32m16/16\u001b[0m  \n",
            "\u001b[?25h\u001b[34m            fastqc\u001b[0m | Found 8 reports\n",
            "\u001b[34m     write_results\u001b[0m | Data        : QC_results/multiqc_data\n",
            "\u001b[34m     write_results\u001b[0m | Report      : QC_results/multiqc_report.html\n",
            "\u001b[34m           multiqc\u001b[0m | MultiQC complete\n"
          ]
        }
      ],
      "source": [
        "# Create directory for trimmed FastQC results\n",
        "!mkdir -p QC_results\n",
        "\n",
        "# Run FastQC on the trimmed FastQ data\n",
        "!fastqc data/*.fastq.gz -o QC_results/\n",
        "\n",
        "# List the trimmed FastQC result files\n",
        "!ls QC_results/\n",
        "\n",
        "# Ejecutar MultiQC\n",
        "!multiqc QC_results/ -o QC_results/"
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "### Visualización de los Resultados de MultiQC\n",
        "\n",
        "Los resultados del análisis de calidad combinada se pueden visualizar desde cualquier navegador web. Para ello, simplemente descarga el informe generado por MultiQC. Este informe está disponible en la ruta `/content/QC_results/multiqc_report.html` en Google Colab.\n",
        "\n",
        "Una vez descargado, abre el archivo `multiqc_report.html` en tu navegador preferido para explorar los resultados detallados del análisis de calidad de tus datos de secuenciación. Este informe incluye gráficos interactivos y secciones detalladas sobre diversos aspectos de la calidad de las lecturas, permitiéndote identificar rápidamente cualquier problema potencial y comparar la calidad entre diferentes conjuntos de datos.\n",
        "\n",
        "### Comentarios Finales\n",
        "\n",
        "Este notebook ha demostrado cómo realizar un análisis de control de calidad exhaustivo de datos de secuenciación utilizando herramientas como FastQC y MultiQC. Hemos cubierto varios pasos clave, incluyendo:\n",
        "\n",
        "- La descarga y visualización de datos FastQ originales.\n",
        "- La extracción de subconjuntos de datos para análisis preliminares.\n",
        "- El trimming de lecturas para mejorar la calidad general de los datos.\n",
        "- La ejecución de análisis de calidad y la generación de informes combinados.\n",
        "\n",
        "Estos procedimientos son fundamentales en la bioinformática para asegurar que los datos de secuenciación sean de alta calidad y adecuados para los análisis posteriores. Al seguir estos pasos, puedes identificar y corregir problemas tempranos en tus datos, lo que mejora la fiabilidad y precisión de tus resultados de investigación.\n",
        "\n",
        "Para más detalles y documentación adicional sobre las herramientas utilizadas, puedes visitar los siguientes enlaces:\n",
        "- [FastQC](https://www.bioinformatics.babraham.ac.uk/projects/fastqc/)\n",
        "- [MultiQC](https://multiqc.info/)\n",
        "- [seqtk](https://github.com/lh3/seqtk)\n",
        "- [fastp](https://github.com/OpenGene/fastp)\n",
        "\n",
        "Espero que este notebook haya sido de ayuda!\n",
        "\n",
        "MEGS"
      ],
      "metadata": {
        "id": "rrI4gt4TG_1A"
      }
    }
  ],
  "metadata": {
    "colab": {
      "provenance": [],
      "authorship_tag": "ABX9TyPxDrstDIE4trrnUylNMT9/",
      "include_colab_link": true
    },
    "kernelspec": {
      "display_name": "Python 3",
      "name": "python3"
    },
    "language_info": {
      "name": "python"
    }
  },
  "nbformat": 4,
  "nbformat_minor": 0
}